llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll

```diff
@@ -0,0 +1,1685 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -mcpu=gfx900 -amdgpu-lower-buffer-fat-pointers < %s | FileCheck %s
+; RUN: opt -S -mcpu=gfx900 -passes=amdgpu-lower-buffer-fat-pointers < %s | FileCheck %s
+
+target datalayout = "e-p:64:64-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-p7:160:256:256:32-p8:128:128-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64-S32-A5-G1-ni:7:8"
```

Comment:
These two are not needed if using `-mtriple=amdgcn`

Reply:
All the existing buffer fat pointer tests explicitly spell out the data layout so I figured I'd follow convention

Reply:
It's rare to have datalayout / triple in the test

Reply:
Should I fix that up here or in a separate bit of NFC?

(And this is testing a pass that needs the layout set correctly)

Reply:
The default layout should be the correct one as long as the triple is set correctly. The explicit data layout specification is only required if one wants to override the default one (for whatever reason).